### PR TITLE
Multi-user problem fixed (i guess so).

### PR DESCRIPTION
**Pull request by Y-Pazniak**
I have added UserProgressStorage to save users data. 
Also I have added UserStorageManager to work with UserProgressStorage.
With every users command I check his command and according to his ID fill the fields of his personal UserProgress object.

### CODE REVIEW

File: src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private UserProgressStorage() {
+        users = new HashMap<>();
+    }
+
+    public static UserProgressStorage getInstance() {
+        return UserProgressStorageHolder.USER_PROGRESS_STORAGE;
+    }
+
+    public UserProgress getUser(String chatId) {
+        if (!users.containsKey(chatId)) {
```
Comment (MaximPilipeiko) [inline src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java:17]: Please, separate get and save on two methods. Do not do it in the same method

File: src/main/java/by/custom/utilcalculator/controller/MessageRouter.java [hunk 2]
```diff
@@ -32,23 +33,25 @@ public SendMessage route(Update update) {
     }
 
     //we receive message from user, check it and handle it if it is ok here
-    public SendMessage getCheckInputMessageAndGetAnswer(Message message) {
+    private SendMessage getCheckInputMessageAndGetAnswer(Message message) {
         String usersMessage = message.getText();
         String userID = message.getChatId().toString();
+        UserProgress userProgress = UserStorageManager.getInstance().getUserProgress(userID);
```
Comment (MaximPilipeiko) [inline src/main/java/by/custom/utilcalculator/controller/MessageRouter.java:39]: Please, move this logic to UserProgressManager. It's not responsibility of router to manage UserProgress

File: src/main/java/by/custom/utilcalculator/controller/UserStorageManager.java
Comment (MaximPilipeiko) [inline src/main/java/by/custom/utilcalculator/controller/UserStorageManager.java:1]: This class is redundant from my view

**Review by MaximPilipeiko — CHANGES_REQUESTED** [review]
Please, rewright it to store UserProgress in file. I think it will help you to understand a conceptual mistake in working with storage.

File: src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java (hunk 1 above)
Comment (Y-Pazniak) [reply src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java:17]: Done

File: src/main/java/by/custom/utilcalculator/controller/MessageRouter.java (hunk 2 above)
Comment (Y-Pazniak) [reply src/main/java/by/custom/utilcalculator/controller/MessageRouter.java:39]: Done.

File: src/main/java/by/custom/utilcalculator/controller/UserStorageManager.java
Comment (Y-Pazniak) [reply src/main/java/by/custom/utilcalculator/controller/UserStorageManager.java:1]: Deleted

**Pull request merged**